src/main/java/mate/project/controller/AuthenticationController.java

```diff
@@ -0,0 +1,25 @@
+package mate.project.controller;
+
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import mate.project.dto.UserRegistrationRequestDto;
+import mate.project.dto.UserResponseDto;
+import mate.project.exception.RegistrationException;
+import mate.project.service.UserService;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/auth")
+public class AuthenticationController {
+    private final UserService userService;
+
+    @PostMapping("/registration")
+    public UserResponseDto register(@Valid @RequestBody UserRegistrationRequestDto request)
+            throws RegistrationException {
```

Comment:
The method 'register' in the controller should not throw exceptions. Instead, handle the exception within the method and return a suitable ResponseEntity as response.